editor/plugins/animation_state_machine_editor.cpp

```diff
@@ -1601,6 +1623,205 @@ void AnimationNodeStateMachineEditor::_update_mode() {
+	}
+
+	for (int i = 0; i < 12; i++) {
+		show_node_group_buttons[i]->set_visible(false);
+		hide_node_group_buttons[i]->set_visible(false);
+		node_group_labels[i]->set_visible(true);
+		node_group_labels[i]->set_text(state_machine->_get_node_group_names()[i]);
+		node_group_name_line[i]->set_visible(false);
+	}
+}
+
+void AnimationNodeStateMachineEditor::_draw_sidepanel_node_group() {
+	PackedStringArray node_group_names = state_machine->_get_node_group_names();
+
+	// !updating is for keeping the name consistent when changing between animation trees while entering a new node group name
+	if ((side_panel_mode == NODE_GROUP_NAME_EDIT) & (!updating)) {
+		for (int i = 0; i < 12; i++) {
+			if (node_group_name_line[i]->get_text() != "") {
+				node_group_names.set(i, node_group_name_line[i]->get_text());
+			}
+		}
+		state_machine->_set_node_group_names(node_group_names);
+	}
+
+	side_panel_mode = NODE_GROUP_SELECTION;
+
+	node_group_names = state_machine->_get_node_group_names();
+
+	for (int i = 0; i < 12; i++) {
+		node_group_name_line[i]->set_placeholder(node_group_names[i]);
+		node_group_labels[i]->set_text(node_group_names[i]);
+	}
+
+	for (int i = 0; i < 12; i++) {
+		show_node_group_buttons[i]->set_visible(true);
+		hide_node_group_buttons[i]->set_visible(true);
+		node_group_name_line[i]->set_visible(false);
+		node_group_select_checkboxes[i]->set_visible(false);
+		node_group_labels[i]->set_visible(true);
+	}
+
+	show_all_node_group_button->set_visible(true);
+	hide_all_node_group_button->set_visible(true);
+	sidepanel_title_2->set_visible(false);
+	node_group_select_menu_button->set_visible(true);
+	node_group_name_edit_menu_button->set_visible(true);
+	node_group_selection_menu_button->set_visible(false);
+	return_to_node_group_selection_menu_button->set_visible(false);
+	sidepanel_title->set_text(TTR("Node Groups"));
+}
+
+void AnimationNodeStateMachineEditor::_draw_sidepanel_node_group_name_edit() {
+	sidepanel_title->set_text(TTR("Node Group Rename"));
+	return_to_node_group_selection_menu_button->set_text(TTR("Save"));
+	side_panel_mode = NODE_GROUP_NAME_EDIT;
+	PackedStringArray node_group_names = state_machine->_get_node_group_names();
+
+	for (int i = 0; i < 12; i++) {
+		show_node_group_buttons[i]->set_visible(false);
+		hide_node_group_buttons[i]->set_visible(false);
+		node_group_labels[i]->set_visible(false);
+		node_group_name_line[i]->set_visible(true);
+		node_group_name_line[i]->set_text("");
+		node_group_name_line[i]->set_placeholder(node_group_names[i]);
+	}
+
+	show_all_node_group_button->set_visible(false);
+	hide_all_node_group_button->set_visible(false);
+	node_group_select_menu_button->set_visible(false);
+	node_group_name_edit_menu_button->set_visible(false);
+	node_group_selection_menu_button->set_visible(false);
+	return_to_node_group_selection_menu_button->set_visible(true);
+}
+
+void AnimationNodeStateMachineEditor::_checkbox_pressed() {
+	if (bool(selected_node) & NODE_GROUP_EDIT) {
```

Comment:
What is this intended to do? It does a bitwise check against a boolean, that doesn't make sense to me

It makes more sense to just do:
```suggestion
	if (selected_node) {
```
As Boolean is just zero or non-zero, it's not guaranteed to be a 1